src/v/cloud_storage/remote_segment.h

```diff
@@ -178,7 +179,7 @@ class remote_segment_batch_reader final {
 
 private:
     friend class single_record_consumer;
-    ss::future<std::unique_ptr<storage::continuous_batch_parser>> init_parser();
```

Comment:
why we need this change? maybe move this change to a separate commit?

Reply:
reverted